tempel.el

```diff
@@ -348,12 +363,30 @@ PROMPT is the optional prompt/default value."
 
 (defun tempel--save ()
   "Save template file buffer."
-  (when-let (buf (get-file-buffer tempel-file))
-    (with-current-buffer buf
-      (when (and (buffer-modified-p) (y-or-n-p (format "Save file %s? " tempel-file)))
-        (save-buffer buf)))))
-
-(defun tempel--file-read (file)
+  (let (modified)
+    (dolist (file (tempel--expand-path tempel-path))
+      (when-let (buffer (find-buffer-visiting file))
+    	(when (buffer-modified-p buffer)
+	  (push buffer modified))))
+    (cond ((not modified))
+	  ((not (cdr modified))
+	   (y-or-n-p "Save template file %s? " (car modified)))
+	  (t
+	   (y-or-n-p "Save modified template files? ")))
+    (mapc #'save-buffer modified)))
+
+(defun tempel--path-modified-p ()
+  "Return non-nil if any template files have been modified since loading."
+  (thread-last (tempel--expand-path tempel-path)
+    (mapcar (lambda (file)
+	      (thread-first file
+		(file-attributes)
+		(file-attribute-modification-time)
+		(time-convert 'integer))))
+    (apply #'max 0)
+    (< tempel--path-modified)))
```

Comment:
Avoid `thread-first` and `thread-last` here. I use these macros myself too occasionally for long pipelines. However three function calls in both cases are not sufficiently many function calls to justify the macro.